proxy/grpcproxy/watcher_group.go

```diff
@@ -63,7 +63,10 @@ func (wg *watcherGroup) broadcast(wr clientv3.WatchResponse) {
 
 	wg.rev = wr.Header.Revision
 	for _, r := range wg.receivers {
-		r.send(wr)
+		if !r.send(wr) {
+			wg.cancel()
```

Comment:
why do we need to cancel the entire watch group here? we should only cancel that single watcher stream connected to one client, right?